src/v/config/bounded_property.h

```diff
@@ -303,7 +304,16 @@ class bounded_property : public property<T> {
                 guess = property<T>::_default;
             }
         }
-        return fmt::format("{}", guess);
+
+        if constexpr (::detail::
+                        is_specialization_of_v<I, std::chrono::duration>) {
+            return fmt::format("{}", guess.count());
+        } else if constexpr (std::is_arithmetic_v<I>) {
+            return fmt::format("{}", guess);
+        } else {
+            static_assert(
+              false, "type does not have a valid example generation strategy");
```

Comment:
i thought i read somewhere that in c++23 or c++26 this will work like this, but afaik we still need to assert on something that is dependent on T (see base::unsupported_type<T>`).

Reply:
This /always/ worked for me, C++17 https://godbolt.org/z/ne1rjnqMs